### PR TITLE
unnessary parenthesis

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 1]
```diff
@@ -391,7 +391,7 @@ Future<Null> diagnoseXcodeBuildFailure(
       // Make sure the user has specified one of:
       // DEVELOPMENT_TEAM (automatic signing)
       // PROVISIONING_PROFILE (manual signing)
-      !(app.buildSettings?.containsKey('DEVELOPMENT_TEAM')) == true
+      !app.buildSettings?.containsKey('DEVELOPMENT_TEAM') == true
           || app.buildSettings?.containsKey('PROVISIONING_PROFILE') == true) {
```
Comment (a14n) [inline packages/flutter_tools/lib/src/ios/mac.dart]: This if statement is really suspect.

1. there's a `!` on nullable
2. the indentation seems to show that there should be parenthesis around the `||`

wdyt ?

btw the line 401 looks also strange. I wonder if there shouldn't be parenthesis around `app.id?.contains('com.example') ?? false` .

Comment (Hixie) [reply packages/flutter_tools/lib/src/ios/mac.dart]: yeah this is super suspect, i was just thinking that to myself before I saw your comment!

cc @cbracken @xster

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/ios/mac.dart (hunk 1 above)
Comment (cbracken) [reply packages/flutter_tools/lib/src/ios/mac.dart]: Yikes. Nice catch. I'll take a quick look.

**Comment by Hixie** [comment]
@a14n feel free to land your patch, we'll take care of the other thing separately. Thanks for catching it.

**Comment by a14n** [comment]
@Hixie ok. I'll wait CIs

btw what would you think about removing unnecessary parenthesis like:

```dart
(a && b) || (c && d)
``` 

(I'm creating a lint to catch unnessecary parenthesis and I wonder if I should limit its action)

**Comment by Hixie** [comment]
Is `||` lower priority than `&&`? So the above is equivalent to `a && b || c && d`?
That seems unobvious. I'd probably keep the parens in that case.

**Comment by a14n** [comment]
https://www.dartlang.org/guides/language/language-tour#operators gives the ordered list of precedence. My lint is based on it.

One advantage about the extended version of the lint is that it learn the user what is the priority list. But I agree that it could be hard to read.

**Comment by Hixie** [comment]
The problem is that different languages have different priority orders and it's not obvious what the order is in any particular language. Based on a quick look at that table I'd probably allow parens around any expression that contains an operator that is then in another expression with another operator of lower (not equal) precedence, except if that operator is `==`/`!=`, `?:`, or any of the assignment operators. (Obviously if the operator is of higher precedence, then the parens are required anyway.)

**Comment by a14n** [comment]
Looking at flutter code base there are both styles. You can find `(a && b) || (c && d)` and `a && b || c && d`.

**Comment by Hixie** [comment]
Yeah, probably depends how familiar the author was with the order of operations at the time. :-) 

Are the cases without parens cases where the `||` is on a line break? I bet that makes the code clearer too...

File: packages/flutter_tools/lib/src/ios/mac.dart (hunk 1 above)
Comment (xster) [reply packages/flutter_tools/lib/src/ios/mac.dart]: Ah man, thanks for the catch. It's totally wrong. It should be 
```dart
!(app.buildSettings?.containsKey('DEVELOPMENT_TEAM') == true
    || app.buildSettings?.containsKey('PROVISIONING_PROFILE') == true)) {
```

Comment (xster) [reply packages/flutter_tools/lib/src/ios/mac.dart]: 401 should have parenthesis like you're saying too.

Comment (cbracken) [reply packages/flutter_tools/lib/src/ios/mac.dart]: #14478 and #14480

**Pull request merged**